### ISSUE TITLE
Some requests on combine service timeout/crash

### ISSUE THREAD

**Issue by bilalshaikh42** [reproducible-biomedical-modeling/CRBM-Viz#2740]
```
   Access to fetch at 'https://combine.api.biosimulations.dev/combine/metadata/biosimulations' from origin 'https://run.biosimulations.dev' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
biosimulations:1 Failed to load resource: the server responded with a status of 504 (Gateway Timeout)
Access to fetch at 'https://combine.api.biosimulations.dev/combine/sedml-specs' from origin 'https://run.biosimulations.dev' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
sedml-specs:1 Failed to load resource: the server responded with a status of 504 (Gateway Timeout)
manifest:1 Failed to load resource: the server responded with a status of 504 (Gateway Timeout)
Access to fetch at 'https://combine.api.biosimulations.dev/combine/manifest' from origin 'https://run.biosimulations.dev' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

**Comment by jonrkarr** [comment]
`https://run.biosimulations.dev` (without the trailing slash) is already in the CORS list for the service:
https://github.com/biosimulations/Biosimulations/blob/dev/apps/combine-service/src/app.py#L24

In my experience, this is working correctly. 

Please post a specific example where this is failing.

**Comment by bilalshaikh42** [comment]
This is not a problem with the python app, but rather the load balancer. it is happening intermittently. I got the alert on the Error logging on google cloud

**Comment by bilalshaikh42** [comment]
@jonrkarr  This one is a bit hard to reliably reproduce. Using this URL tends to trigger it, but not always: Sometimes this causes CORS errors, and other times 504 errors.

 https://run.biosimulations.org/run?projectUrl=https:%2F%2Frun.api.biosimulations.org%2Frun%2F60f906bc227eb28472089ccb%2Fdownload&simulator=bionetgen&simulatorVersion=2.6.0&runName=Dolan%202020:%20Non-homologous%20end%20joining%20(BNGL,%20NFSim,%20BioNetGen)%20(rerun)

here is the only logs: 
```
30
[2021-07-30 17:05:26 +0000] [9] [INFO] Starting gunicorn 20.1.0
29
[2021-07-30 17:05:26 +0000] [9] [INFO] Listening at: http://0.0.0.0:3333 (9)
28
[2021-07-30 17:05:26 +0000] [9] [INFO] Using worker: sync
27
[2021-07-30 17:05:26 +0000] [18] [INFO] Booting worker with pid: 18
26
[2021-07-30 17:05:26 +0000] [10] [INFO] Starting gunicorn 20.1.0
25
[2021-07-30 17:05:26 +0000] [10] [INFO] Listening at: http://0.0.0.0:3333 (10)
24
[2021-07-30 17:05:26 +0000] [10] [INFO] Using worker: sync
23
[2021-07-30 17:05:26 +0000] [19] [INFO] Booting worker with pid: 19
22
[2021-07-30 17:05:38 +0000] [9] [INFO] Starting gunicorn 20.1.0
21
[2021-07-30 17:05:38 +0000] [9] [INFO] Listening at: http://0.0.0.0:3333 (9)
20
[2021-07-30 17:05:38 +0000] [9] [INFO] Using worker: sync
19
[2021-07-30 17:05:38 +0000] [18] [INFO] Booting worker with pid: 18
18
[2021-07-30 17:06:44 +0000] [9] [CRITICAL] WORKER TIMEOUT (pid:18)
17
[2021-07-30 17:06:44 +0000] [18] [INFO] Worker exiting (pid: 18)
16
[2021-07-30 17:06:44 +0000] [10] [CRITICAL] WORKER TIMEOUT (pid:19)
15
[2021-07-30 17:06:44 +0000] [9] [CRITICAL] WORKER TIMEOUT (pid:18)
14
[2021-07-30 17:06:44 +0000] [18] [INFO] Worker exiting (pid: 18)
13
[2021-07-30 17:06:44 +0000] [20] [INFO] Booting worker with pid: 20
12
[2021-07-30 17:06:44 +0000] [19] [INFO] Worker exiting (pid: 19)
11
[2021-07-30 17:06:45 +0000] [20] [INFO] Booting worker with pid: 20
10
[2021-07-30 17:06:45 +0000] [21] [INFO] Booting worker with pid: 21
9
[2021-07-30 17:07:20 +0000] [9] [CRITICAL] WORKER TIMEOUT (pid:20)
8
[2021-07-30 17:07:20 +0000] [20] [INFO] Worker exiting (pid: 20)
7
[2021-07-30 17:07:21 +0000] [22] [INFO] Booting worker with pid: 22
6
[2021-07-30 17:07:23 +0000] [10] [CRITICAL] WORKER TIMEOUT (pid:21)
5
[2021-07-30 17:07:23 +0000] [21] [INFO] Worker exiting (pid: 21)
4
[2021-07-30 17:07:24 +0000] [23] [INFO] Booting worker with pid: 23
3
[2021-07-30 17:07:26 +0000] [9] [CRITICAL] WORKER TIMEOUT (pid:20)
2
[2021-07-30 17:07:26 +0000] [20] [INFO] Worker exiting (pid: 20)
1
[2021-07-30 17:07:26 +0000] [22] [INFO] Booting worker with pid: 22
```
It seems that the actual flask logs are not outputting anything, but the requests time-out, causing the service to fail and return 504.

**Comment by jonrkarr** [comment]
I've occasionally seen something like this locally. Not sure where to debug.

Is this something we could partly address with more workers?

**Comment by bilalshaikh42** [comment]
I am not sure if the timeouts are happening for one request or multiple

**Comment by bilalshaikh42** [comment]
I think this may be due to not being able to connect to the s3 bucket

**Comment by jonrkarr** [comment]
That meshes with what I've experienced locally. Locally, I haven't been paying attention to S3 access. I haven't necessarily set the environment variable required for S3 access.

**Comment by bilalshaikh42** [comment]
Taking a look at this: https://serverfault.com/questions/490101/how-to-resolve-the-gunicorn-critical-worker-timeout-error

Do some of the calls take more than 30 seconds? We should be able to tweak the docker image to change the timeout to be a bit higher.  We should only do this if the computation itself takes that long sometimes. If this is because of a slow connection to the s3 bucket, then we should leave it so we get back http error sooner rather than later. We can check to see if this pops up again after [connection issue](https://status.biosimulations.org/incident/2) is fixed.

**Comment by jonrkarr** [comment]
Validating metadata can be longer because this is setup to use the identifiers.org API to check URIs. The other endpoints should complete quickly.

**Comment by jonrkarr** [comment]
I sped up the validation of Identifiers.org URIs in metadata by downloading the database and doing the validation locally rather than with individual REST queries for each URI. See #2757.

**Comment by bilalshaikh42** [comment]
This seems to be resolved with the change in #2757  and the resolution of  https://status.biosimulations.org/incident/2
